### PR TITLE
offsetfile: Buffer the leveldb, reader, and writer for offestfile

### PR DESCRIPTION
**Pull request by kcalvinalvin**
Offsetfile building but now with buffers. In my machine, sys and user time is 80% less but the wall clock is only showing 30% less time :/ I guess there's some serious disk io bottleneck going on that *could* be improved with more effort. Just not sure if it's worth the effort.

Benchmark added in offestfile_test.go and also fixes a test in offsetfile_test.go which was broken in master. We could also start to use the tests here...

### CODE REVIEW

**Comment by dergoegge** [comment]
I ran the tests. it works and it it's significantly faster!

**Comment by dergoegge** [comment]
also tested with an external HDD, my internal SSD is about 10x faster but before buffering it took for ever on the HDD.

**Comment by kcalvinalvin** [comment]
Merged #151

File: bridgenode/offsetfile.go [hunk 1]
```diff
@@ -48,14 +48,18 @@ func buildOffsetFile(dataDir string, tip util.Hash,
 		}
 	}
 
-	lvdb, err := OpenIndexFile(dataDir)
-	if err != nil {
-		panic(err)
-	}
-	defer lvdb.Close()
+	lvdb := OpenIndexFile(dataDir)
```
Comment (adiabat) [inline bridgenode/offsetfile.go:51]: Doesn't catch the error...
actually I think I can merge anyway and just put it in

**Comment by adiabat** [comment]
Looks good!
(Well actually doesn't compile but that's a rebase thing, works fine with that)

**Pull request merged**